imagersite/imagerprofile/views.py

```diff
@@ -1,3 +1,31 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
+from django.contrib.auth.models import User
+from .forms import ProfileForm
 
-# Create your views here.
+
+def all_profiles_page(request):
+    users = User.objects.filter(is_active=True).all()
+    return render(request, 'profilelist.html', context={'users': users})
+
+
+def profile_page(request, user_id=1):
+    try:
+        user = User.objects.get(id=user_id)
+    except User.DoesNotExist:
+        return render(request, 'error.html',
+                      context={'error_type': '404 Not Found'})
```

Comment:
Django actually has a 404 view that you would do better to override, then here you simply raise HttpNotFound.  The problem with doing it this way is that although the page will *say* 404, the HTTP response code will be `200 OK`, which harms the machine readability of your website.